### ISSUE TITLE
Remove URL repositories

### ISSUE THREAD

**Issue by jasontedor** [elastic/elasticsearch#72197]
This issue is to capture our thinking around removing URL repositories, and therefore solicit feedback from the community. Assuming no unforeseen blockers, we plan to deprecate URL repositories and remove them in a future release of Elasticsearch.

URL repositories are read-only repositories to provide access to snapshots for restore purposes. For example, this can be useful for sharing data. However, such use-cases can also be served by storing snapshots in a public bucket in a Cloud storage provider, and access the snapshot via HTTP. Additionally, all repositories can be mounted with a `read_only` property enabled, and we think that providing read-only access to a repository is better handled through authorization than a specific repository type.

Another use-case for URL repositories that has been mentioned is for providing access to a snapshot stored on a shared filesystem fronted by an HTTP proxy. For such use-cases, these days there are many simple ways of standing up an S3 proxy in front of a shared filesystem (e.g., MinIO). With this, access to the snapshot can be provided by using the S3 plugin, instead of using a URL repository.

In summary, we think the existing use-cases for URL repositories are better served by other repository types. We would like to remove URL repositories to simplify our maintenance burden.

We welcome feedback on this proposal, including any use-cases that we might have overlooked.

**Comment by nik9000** [comment]
Sounds like no one's replied. Does that mean folks rarely use them?

I've added team-discuss to get this back on the team's radar.

**Comment by jasontedor** [comment]
@nik9000 We already discussed this within the distributed team a couple of months ago (late April, early May). Our conclusion at the time was that URL repositories could probably be removed, and our action item was to open an issue to capture our thinking, and solicit feedback.

Seeing as we have none, I think we can safely conclude that we can remove URL repositories.

**Comment by jasontedor** [comment]
By the way, the other way that we measured lack of use of them was by looking at bug counts, discuss issues, and support cases involving them, all of which indicated very little use.
